### PR TITLE
Hide status based on custom function

### PR DESCRIPTION
**Pull request by antoineco**
#### tl;dr

Hide kube-ps1 when the `KUBE_PS1_HIDE_FUNCTION` function is defined and returns `0`.

#### Description

I find it quite handy to be able to disable prompt modules selectively, depending on the context of my workspace. For example, I have no reason to be displaying a Kubernetes status when I'm currently not working on a cluster.

This PR adds a very simple mechanism to allow users to _hide_ the kube-ps1 status based on the return value of a custom function. I used the term _hide_ instead of _disable_ because `_kube_ps1_update_cache()` is still being invoked.

#### Examples

A few basic but typical use-cases come to my mind:

```bash
# Hide kube-ps1 when the current context is "minikube".
hide_kube_ps1() {
    [[ "$KUBE_PS1_CONTEXT" == 'minikube' ]]
}
```

```bash
# Hide kube-ps1 when KUBECONFIG is unset.
hide_kube_ps1() {
    [[ -z "${KUBECONFIG:-}" ]]
}
```

```bash
# Hide kube-ps1 when the current cluster is unreachable.
hide_kube_ps1() {
    ! kubectl cluster-info --request-timeout=1s &>/dev/null
}
```

```bash
# Hide kube-ps1 when on a given network (macOS).
hide_kube_ps1() {
    route -n get default | awk '$1 == "gateway:" && $2 !~ /^192\.168\.100\./ { exit 1 }'
}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -132,8 +145,10 @@ the following environment variables:
 | `KUBE_PS1_SEPARATOR` | &#124; | Separator between symbol and cluster name |
 | `KUBE_PS1_DIVIDER` | `:` | Separator between cluster and namespace |
 | `KUBE_PS1_SUFFIX` | `)` | Prompt closing character |
-| `KUBE_PS1_CLUSTER_FUNCTION` | No default, must be user supplied | Function to customize how cluster is displayed |
-| `KUBE_PS1_NAMESPACE_FUNCTION` | No default, must be user supplied | Function to customize how namespace is displayed |
+| `KUBE_PS1_CLUSTER_FUNCTION` | (none)\* | Function to customize how cluster is displayed |
+| `KUBE_PS1_NAMESPACE_FUNCTION` | (none)\* | Function to customize how namespace is displayed |
+
+\* _must be user supplied_
```
Comment (antoineco) [inline README.md]: _note_: Not directly related to this PR but I found the middle column unnecessarily large. Happy to revert if you like the full test better.

**Comment by antoineco** [comment]
@jonmosco friendly ping 👈  Do you have any thoughts about that feature? If you don't like it no hard feeling, I can keep a fork on the side.

**Comment by jonmosco** [comment]
Im going to pass on this function.  Thank you, I just want to keep it as vanilla as possible.
